### PR TITLE
ipq40xx: Add support for CIG WF-610D AP

### PR DESCRIPTION
**Pull request by chaitanyagodavarthi**
Add support for CIG WF-610D AP

Signed-off-by: Chaitanya Godavarthi <chaitanya.kiran@netexperience.com>

### CODE REVIEW

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch [hunk 1]
```diff
@@ -0,0 +1,691 @@
...
++	cig,wf610d)
++		cig_do_upgrade "$1"
++		;;
++
+ 	*)
+ 		default_do_upgrade "$1"
+ 		;;
+diff --git a/target/linux/ipq40xx/config-4.14 b/target/linux/ipq40xx/config-4.14
+index c72e693206..29c500e533 100644
+--- a/target/linux/ipq40xx/config-4.14
```
Comment (blogic) [inline patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:154]: I dont think any of these are .config changes are required

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch [hunk 2]
```diff
@@ -0,0 +1,691 @@
...
++	qcom,msm-id = <0x111 0>;
++	qcom,pmic-id = <0 0 0 0>;
++
++	memory {
++		device_type = "memory";
++		reg = <0x80000000 0x10000000>; /* 256MB */
++	};
++
++	chosen {
++		bootargs-append = " clk_ignore_unused root=/dev/mtdblock9 debug ignore_loglevel loglevel=8 break=y log_buf_len=10M print_fatal_signals=1";
```
Comment (blogic) [inline patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:287]: are these required ? looks like copy&paste

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch [hunk 3]
```diff
@@ -0,0 +1,691 @@
...
++			pinctrl-0 = <&mdio_pins>;
++			pinctrl-names = "default";
++			phy-reset-gpio = <&tlmm 18 0>;
++			status = "okay";
++			bias-disable;
++		};
++
++		ess-switch@c000000 {
++			status = "okay";
++//			switch_mac_mode = <0x0>; /* mac mode for RGMII RMII */
```
Comment (blogic) [inline patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:540]: remove dead code please

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch [hunk 4]
```diff
@@ -0,0 +1,691 @@
...
++			status = "okay";
++		};
++
++		usb2: usb2@60f8800 {
++			status = "okay";
++		};
++
++		ext_wtd {
++			compatible = "linux,wdt-gpio";
++			gpios = <&tlmm 41 GPIO_ACTIVE_LOW>;
```
Comment (blogic) [inline patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:611]: ah, there was indeed a hw WDT ?

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch [hunk 5]
```diff
@@ -0,0 +1,691 @@
...
++		};
++
++		tcsr@1957000 {
++			compatible = "qcom,tcsr";
++			reg = <0x1957000 0x100>;
++			qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
++		};
++
++		wifi0: wifi@a000000 {
++			//wifi_led_num = <2>;         // Wifi 2G
```
Comment (blogic) [inline patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:630]: remove dead code please

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch [hunk 6]
```diff
@@ -0,0 +1,691 @@
...
+diff --git a/target/linux/ipq40xx/base-files/etc/board.d/01_leds b/target/linux/ipq40xx/base-files/etc/board.d/01_leds
+index 5a36b642e9..9d3a51f7ea 100755
+--- a/target/linux/ipq40xx/base-files/etc/board.d/01_leds
++++ b/target/linux/ipq40xx/base-files/etc/board.d/01_leds
+@@ -64,6 +64,11 @@ zyxel,wre6606)
+ 	ucidef_set_led_wlan "wlan2g" "WLAN2G" "${boardname}:green:wlan2g" "phy0tpt"
+ 	ucidef_set_led_wlan "wlan5g" "WLAN5G" "${boardname}:green:wlan5g" "phy1tpt"
+ 	;;
++cig,wf610d)
++	ucidef_set_led_wlan "wlan2g" "WLAN2G" "${boardname}:blue:wlan2g" "phy0tpt"
```
Comment (blogic) [inline patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:50]: this block is identical to the one above, please just ass the cig identifier to the switch clause

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch [hunk 7]
```diff
@@ -0,0 +1,691 @@
...
+ esac
+diff --git a/target/linux/ipq40xx/base-files/etc/board.d/02_network b/target/linux/ipq40xx/base-files/etc/board.d/02_network
+index 56de6e61c3..33cc71296e 100755
+--- a/target/linux/ipq40xx/base-files/etc/board.d/02_network
++++ b/target/linux/ipq40xx/base-files/etc/board.d/02_network
+@@ -69,6 +69,11 @@ ipq40xx_setup_interfaces()
+ 		ucidef_add_switch "switch0" \
+ 			"0u@eth0" "3:lan" "4:lan" "0u@eth1" "5:wan"
+ 		;;
++	cig,wf610d)
```
Comment (blogic) [inline patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:65]: Is there no existing block with that setup ?

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch [hunk 8]
```diff
@@ -0,0 +1,691 @@
...
+diff --git a/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata b/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+index ff77aed304..d36fc505ff 100644
+--- a/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
++++ b/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+@@ -199,6 +199,15 @@ case "$FIRMWARE" in
+ 	tp-link,ec420-g1)
+ 		ath10kcal_extract "0:ART" 4096 12064
+ 		ath10kcal_patch_mac_crc $(macaddr_add "$(get_tip mac_address)" 9) #2.4G, wlan1
++		;;
++	cig,wf610d)
```
Comment (blogic) [inline patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:82]: again, this is generic ART behaviour, please use one of the existing code paths

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch [hunk 9]
```diff
@@ -0,0 +1,691 @@
...
+@@ -1,3 +1,4 @@
++. /lib/functions/system.sh
+ PART_NAME=firmware
+ REQUIRE_IMAGE_METADATA=1
+ 
+@@ -44,6 +45,20 @@ zyxel_do_upgrade() {
+ 	fi
+ }
+ 
++cig_do_upgrade() {
```
Comment (blogic) [inline patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:124]: also here, There is an exisitng code path to flash these units. please make use of it

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch (hunk 1 above)
Comment (chaitanyagodavarthi) [reply patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:154]: Removed the unnecessary ones

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch (hunk 4 above)
Comment (chaitanyagodavarthi) [reply patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:611]: Yes, there was, we came to know that after enquiry with them.

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch (hunk 9 above)
Comment (chaitanyagodavarthi) [reply patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:184]: Hi, the only relevant path I find is nand_upgrade_tar, but it searches for Ubi partitions. which we don't have.

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch [hunk 10]
```diff
@@ -0,0 +1,639 @@
...
++CONFIG_GPIO_WATCHDOG_ARCH_INITCALL=y
+ # CONFIG_GRO_CELLS is not set
+ CONFIG_HANDLE_DOMAIN_IRQ=y
+ CONFIG_HARDEN_BRANCH_PREDICTOR=y
+@@ -251,6 +251,8 @@ CONFIG_I2C_CHARDEV=y
+ CONFIG_I2C_HELPER_AUTO=y
+ CONFIG_I2C_QUP=y
+ CONFIG_INITRAMFS_SOURCE=""
++CONFIG_INPUT=y
++CONFIG_INPUT_KEYBOARD=y
```
Comment (blogic) [inline patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:224]: do we really have a keyboard ?

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch [hunk 11]
```diff
@@ -0,0 +1,639 @@
...
++cig_do_upgrade() {
++	local tar_file="$1"
++
++	local board_dir=$(tar tf $tar_file | grep -m 1 '^sysupgrade-.*/$')
++	board_dir=${board_dir%/}
++	local dev=$(find_mtd_chardev "0:HLOS")
++
++	tar Oxf $tar_file ${board_dir}/kernel | mtd write - ${dev}
++
++	tar Oxf $tar_file ${board_dir}/root | mtd write - rootfs
```
Comment (blogic) [inline patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:187]: this is still not using the generic upgrade code.

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch (hunk 10 above)
Comment (chaitanyagodavarthi) [reply patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:224]: We need it for gpio key

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch (hunk 11 above)
Comment (blogic) [reply patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:187]: that is why we have gpio-button-hotplug. please use that instead

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch (hunk 10 above)
Comment (blogic) [reply patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:224]: that is why we have gpio-button-hotplug. please use that instead

File: patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch [hunk 12]
```diff
@@ -0,0 +1,647 @@
...
++	qxwlan,e2600ac-c2 |\
++	cig,wf610d)
+ 		ath10kcal_extract "0:ART" 20480 12064
+ 		;;
+ 	engenius,ens620ext)
+@@ -266,6 +268,7 @@ case "$FIRMWARE" in
+ 		;;
+ 	esac
+ 	;;
++
```
Comment (blogic) [inline patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch:159]: unrelated change, please remove it

**Comment by blogic** [comment]
merged into pending, Thanks !
